client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -345,4 +358,66 @@ public void close() {
             this.connectionPool.close();
         }
     }
+
+    @Override
+    public CompletableFuture<Long> getDistanceBetweenTwoStreamCuts(Stream stream, StreamCut fromStreamCut, StreamCut toStreamCut) {
+        Preconditions.checkNotNull(fromStreamCut, "fromStreamCut");
+        Preconditions.checkNotNull(toStreamCut, "toStreamCut");
+        if (!fromStreamCut.equals(StreamCut.UNBOUNDED) && !toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            Preconditions.checkArgument(fromStreamCut.asImpl().getStream().equals(toStreamCut.asImpl().getStream()),
+                    "Ensure streamCuts for the same stream is passed");
+        }
+        final CompletableFuture<StreamSegmentSuccessors> unread;
+        final Map<Segment, Long> endPositions;
+        if (!fromStreamCut.equals(StreamCut.UNBOUNDED) && toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            CompletableFuture<StreamInfo> streamInfo = fetchStreamInfo(stream.getScope(), stream.getStreamName());
+            StreamCut endSC = streamInfo.join().getTailStreamCut();
+            unread = controller.getSegments(fromStreamCut, endSC);
+            endPositions = endSC.asImpl().getPositions();
+        } else if (fromStreamCut.equals(StreamCut.UNBOUNDED) && !toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            StreamCut startSC = getStartStreamCutForUnboundedSC(stream);
+            unread = controller.getSegments(startSC, toStreamCut);
+            endPositions = toStreamCut.asImpl().getPositions();
+        } else {
+            unread = controller.getSegments(fromStreamCut, toStreamCut);
```

Comment:
Does this branch work for the case where both are unbounded? Or should it go through both of the above branches to get a new start and end cut?

Reply:
currently for Unbounded case it just sets an empty map, so we handled that in the above cases. I have added a new condition when both are unbounded.